spec/unit/facter/simplib__sshd_config_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'spec_helper'
+
+describe "custom fact simplib__sshd_config" do
+
+  before(:each) do
+    Facter.clear
+
+  end
+
+  context 'with a simp /etc/ssh/sshd_config' do
+    let(:sshd_config_content) { <<-EOM
+#Brief chunk of file 
+Port 22
+ListenAddress 0.0.0.0
+#ListenAddress ::
+
+#PubkeyAuthentication yes
+
+# The default is to check both .ssh/authorized_keys and .ssh/authorized_keys2
+# but this is overridden so installations will only check .ssh/authorized_keys
+AuthorizedKeysFile      /etc/ssh/local_keys/%u
+
+#AuthorizedPrincipalsFile none
+
+#AuthorizedKeysCommand none
+AuthorizedKeysCommand /usr/bin/sss_ssh_authorizedkeys
+#AuthorizedKeysCommandUser nobody
+AuthorizedKeysCommandUser nobody
+
+# Even inline comments!
+    # And indented comments
+      EOM
+    }
+    it 'should return hash of values from /etc/ssh/sshd_config' do
+      File.expects(:exist?).with('/etc/ssh/sshd_config').returns(true)
+      File.expects(:readable?).with('/etc/ssh/sshd_config').returns(true)
+      File.expects(:read).with('/etc/ssh/sshd_config').returns(sshd_config_content)
+
+      # This resets the stubbing code in Mocha to ensure that the code does not
```

Comment:
should we be bothering to make a work around for Mocha, when it is deprecated? (spits out message about it when we run the tests)

Reply:
Mocha isn't deprecated; the warning is just to draw attention to the user that puppetlabs_spec_helper defaults to mocha.

Reply:
ahhh